src/state.py

```diff
@@ -873,3 +880,31 @@ def _parse_script_url(charm: ops.CharmBase) -> str | None:
     if not parsed_url.scheme or not parsed_url.hostname:
         raise InvalidScriptURLError("Invalid script URL, must contain scheme and hostname.")
     return script_url_str
+
+
+class InvalidSecretError(CharmConfigInvalidError):
+    """Represents an error when fetching secrets."""
+
+
+def _parse_script_secrets(charm: ops.CharmBase) -> dict[str, str] | None:
+    """Parse secrets to load as environment variables for the external script.
+
+    Args:
+        charm: The running charm instance.
+
+    Raises:
+        InvalidSecretError: If a secret of invalid format (secrets separated by space)
+    """
+    script_secret_label = typing.cast(str, charm.config.get(SCRIPT_SECRET_LABEL_CONFIG_NAME, ""))
+    if not script_secret_label:
+        return None
+    try:
+        secret = charm.model.get_secret(id=SCRIPT_SECRET_LABEL_CONFIG_NAME)
```

Comment:
Hm here you're fetching the secret by `id` and not by `label` and the argument is set from the constant and not from the config option (should use `script_secret_label`)

Reply:
Due to the limitations of Juju 3.1, I've added an extra option to setup secret as a config option.
Also fixed the issue with reading secrets via ID, thank you for the review :)